### ISSUE TITLE
Flaky test_backup_restore_on_cluster/test_disallow_concurrency.py

### ISSUE THREAD

**Issue by SmitaRKulkarni** [ClickHouse/ClickHouse#45486]
This test is to check if disallowing of concurrent backups & restores works as expected. 
When one backup/restore fails, the status is not updated properly, thus we need to set the status of backup/restore so that other nodes in cluster are aware of failures.

Some occurences : 
https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksICBncm91cFVuaXFBcnJheShwdWxsX3JlcXVlc3RfbnVtYmVyKSwgIGFueShyZXBvcnRfdXJsKQpmcm9tIGNoZWNrcyB3aGVyZSAnMjAyMi0wNi0wMScgPD0gY2hlY2tfc3RhcnRfdGltZSBhbmQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X2JhY2t1cF9yZXN0b3JlX29uX2NsdXN0ZXIvdGVzdF9kaXNhbGxvd19jb25jdXJyZW5jeS5weSUnIGFuZCB0ZXN0X3N0YXR1cyBpbiAoJ0ZBSUwnLCAnRkxBS1knKSBncm91cCBieSBkIG9yZGVyIGJ5IGQgZGVzYw==

**Comment by vdimir** [comment]
`test_backup_restore_on_cluster` also fails with ASan asserts:

https://play.clickhouse.com/play?user=play#c2VsZWN0IApjaGVja19zdGFydF90aW1lLAogICAgcmVwb3J0X3VybCwgY291bnQoKQpGUk9NIGNoZWNrcwpXSEVSRSB0ZXN0X25hbWUgbGlrZSAnJXRlc3RfYmFja3VwX3Jlc3RvcmVfb25fY2x1c3RlciUnIApBTkQgdGVzdF9zdGF0dXMgaW4gKCdGQUlMJywgJ0ZMQUtZJykKQU5EIHB1bGxfcmVxdWVzdF9udW1iZXIgPT0gMApHUk9VUCBCWSBjaGVja19uYW1lLCByZXBvcnRfdXJsLCBjaGVja19zdGFydF90aW1lCm9yZGVyIGJ5IGNoZWNrX3N0YXJ0X3RpbWUgZGVzYwpMSU1JVCAyMA==

**Comment by rschu1ze** [comment]
@SmitaRKulkarni This issue popped up in tests of #46075 ([here](https://s3.amazonaws.com/clickhouse-test-reports/46075/3259353bd3f41a59b73b9072dc0e7a862930c00b/integration_tests__release__[3/4].html)), error message:
```
>       assert "Concurrent backups not supported" in nodes[2].query_and_get_error(
            f"BACKUP TABLE tbl ON CLUSTER 'cluster' TO {backup_name}
```

Would you like to check if this is a new issue or this one? Thanks!

**Comment by SmitaRKulkarni** [comment]
@rschu1ze  : Looks like it will be resolved after this #45982. will keep an eye on the integration test fails

**Comment by tavplubix** [comment]
It's still flaky 
https://s3.amazonaws.com/clickhouse-test-reports/47863/73ffdd893dccfe29e8b04a464bba33bf72b6d95e/integration_tests__asan__[3/6].html
